Refactor Prime (#14)

Alexangelj opened the pull request:
Fixes (#14)

No state in the lib files. Still needs to review the flow of functions. Functions like handleOrder in Prime.jsx are called from OpenPosition component, which then call the handleOrder function in the lib file exchange.js

BrockSmedley commented on src/lib/Erc20.js:
```diff
@@ -0,0 +1,118 @@
+
+import PrimeContract from '../artifacts/Prime.json';
+
+const handleMint = async (web3, symbol) => {
```
In the lib functions, can you add doc-comments? This makes it much easier (for me, at least) to quickly figure out what a function does and what to expect as far as params and return values go.

```
/**
 * High-level explanation of what this function does.
 * @param {*} web3 a web3 object.
 * @param {*} id high-level description of this identifier.
 * 
 * @returns Some shit.
 */
const someFunction = async (web3, id) => { ... };
```

Here's an intro to doc-comments (JSDoc) but you probably don't need it.
https://jsdoc.app/about-getting-started.html

If you have the proper VSCode plugins installed, all you need to do is type `/**` above a function and it will auto-fill.

BrockSmedley commented on src/lib/option.js:
```diff
@@ -0,0 +1,493 @@
+        'put': {},
+    }
+
+    async function getUnfilledBuyOrders(instance) {
```
It looks like `getOptions` contains function definitions... Can you move those out of the function? You don't need to export them (obviously) but it's pretty hard to read a function that has other functions inside of it. Really, the only reason to do that is if you have a super-specific lambda that you need to use a lot, and only in that function. An example might be `const someSpecificCalculation = (x, y) => { Math.sqrt(x^^y / 2 * some_locally_scoped_variable) };`

But never anything that spans multiple lines.

BrockSmedley commented on src/lib/option.js:
```diff
@@ -0,0 +1,493 @@
+    return optionSelection;
+};
+
+/* TEMP FUNCTION - DONT NEED */
```
Can this be deleted?

BrockSmedley commented on src/components/accountOrders/accountOrders.jsx:
```diff
@@ -12,20 +12,33 @@ import TableContainer from '@material-ui/core/TableContainer';
+    }
+
+    componentDidMount = async () => {
+        console.log('account order web3', this.props.web3)
```
at some point we should remove all instances of `console.log`

BrockSmedley commented on src/components/optionsChainTable/optionsChainTable.jsx:
```diff
@@ -32,17 +67,20 @@ class OptionsChainTable extends Component {
-        if (expiration) {
-            date = new Date(expiration * 1000);
+        if (this.props.expiration) {
+            date = new Date(this.props.expiration * 1000);
```
Is this 1000 for converting from ms to seconds? Might be more readable to have a globally available constant called `SECONDS` that is equal to 1000. That way you can write `expiration * SECONDS`.

BrockSmedley commented on src/components/optionsChainTable/selectOptionsChain.jsx:
```diff
@@ -0,0 +1,49 @@
+                        (pair) ? props.classes.rowButtonS : props.classes.rowButtonL
+                    }
+                    onClick={
+                        (pair)
```
Does `pair` have to be inside parenthesis, or can we omit them? (I'm not a JS expert).

BrockSmedley commented on src/components/optionsChainTable/selectOptionsChain.jsx:
```diff
@@ -0,0 +1,49 @@
+                </Button>
+                <Button
+                    className={
+                        (expiration) ? props.classes.rowButtonS : props.classes.rowButtonL
```
same as above for all these guys with parenthesis

BrockSmedley commented on src/lib/account.js:
```diff
@@ -0,0 +1,114 @@
+    }
+
+    let chain;
+    let id;
```
👍 good use of let

BrockSmedley commented on src/lib/exchange.js:
```diff
@@ -0,0 +1,320 @@
+
+            /* await this.getPendingTx(false); */
+        } catch (error) {
+            /* await this.getPendingTx(false); */
```
Why are these commented?

BrockSmedley replied:
If something needs to be done but is outside the scope of this PR, we should write a `TODO` comment so that we can easily CTRL-F any loose ends.

Alexangelj replied:
The function getPendingTx was acting as my animation loading for the order form openPosition.jsx. I wasn't sure how we wanted to handle loading data so I commented it out. I will leave a 'FIX' there, or I can remove them.

BrockSmedley commented on src/lib/option.js:
```diff
@@ -0,0 +1,493 @@
+    const _pow = expiration;
+    const _gem = account;
+
+    /* result = await primeInstance.methods.createPrime(
```
Why is this commented? Can we delete?

Alexangelj commented:
Added a new commit that resolves all the comments you made. Still room for improvement

Alexangelj commented:
Added a new commit that refactors option.js, specifically the `getOptions` function. I've split it into multiple parts that call functions into other lib files. Hopefully is much more readable now.

Alexangelj commented:
In commit #5 I moved some web3 specific functions from lib/utils to lib/web3. lib/web3 will have all the web3 specific functions like `fromWei` and instantiating new contract instances.

I also added all the available functions that are  callable in Pool.sol to pool.js. I added more callable functions in Prime.sol to prime.js.

BrockSmedley commented on src/components/accountOrders/accountOrders.jsx:
```diff
@@ -1,31 +1,40 @@
+
+    componentDidMount = async () => {
+        if(this.props.web3) {
+            const accountOrders = await getAccountOrders(this.props.web3);
```
Component-specific state! Awesome 👍

BrockSmedley commented on src/components/optionsChainTable/optionsChainTable.jsx:
```diff
@@ -32,17 +68,20 @@ class OptionsChainTable extends Component {
-        if (expiration) {
-            date = new Date(expiration * 1000);
+        if (this.props.expiration) {
+            date = new Date(this.props.expiration * SECONDS);
```
yeet

BrockSmedley commented on src/lib/Erc20.js:
```diff
@@ -0,0 +1,165 @@
+    return mint;
+};
+
+/**
```
doc-comments FTW
